Base/BacnetPrescale.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System.IO.BACnet.Serialize;
+
+namespace System.IO.BACnet
+{
+    public class BacnetPrescale : ASN1.IEncode, ASN1.IDecode
+    {
+        public uint Multiplier
+        {
+            get { return multiplier; }
+            set { multiplier = value; }
+        }
+
+        public uint Modulo
+        {
+            get { return modulo; }
+            set { modulo = value; }
+        }
+
+        public uint multiplier;
+        public uint modulo;
```

Comment:
I know this is how the code base is written, but maybe we could start using automatic properties? The problem is that in many places these values are being used with ```ref``` or ```out``` so only having properties wouldn't work, but then why do we need properties here in the first place? Just saying out loud, this can remain as is because it's consistent with the existing codebase.

Reply:
I am not good enough with C# to give suggestions, but if you wish to redesign the whole ASN.1 so that it's cleaner, I'm down with helping out.

Reply:
I'm happy to hear that, all help is appreciated! Learning C# tricks is easy and I can suggest that in code reviews. Being alergic to code smells like code duplications, or spaghetti code, that's something I'm looking forward to! I think once we eliminate them the code base will be way lighter and less error prone. You can already see on v2 branch that only eleminating duplications in mainly ```BacnetClient.cs``` resulted in removing 600 lines. Let me know which part you would like to fix/extend and I will let you know how I would see it. You can open an issue so we can discuss it there.

Reply:
This is only somewhat related to ASN.1 but I would love it if the user can override the creation of the BacnetXXX classes during a decode with a user-defined factory. Why I want this is so that I can extend the classes with my own, for overriding ToString or adding serialization/deserialization to proprietary format when saving BACnet data that we received from the stack.

I am not sure how the factory should be passed to the stack, whether it should be a global/static one or separate per property type.